lib/js-preprocessor/broccoli-generate-components.js

```diff
@@ -0,0 +1,75 @@
+/* jshint node: true */
+'use strict';
+
+var fs = require('fs');
+var path = require('path');
+
+var CachingWriter = require('broccoli-caching-writer');
+var walkSync = require('walk-sync');
+var existsSync = require('exists-sync');
+var mkdirp = require('mkdirp');
+
+GenerateComponents.prototype = Object.create(CachingWriter.prototype);
+GenerateComponents.prototype.constructor = GenerateComponents;
+
+function GenerateComponents(inputTree, options) {
+  CachingWriter.call(this, [inputTree]);
+
+  this.registryType = options.registryType;
+  this.moduleName = options.moduleName;
+  this.prefix = options.prefix;
+  this.baseDir = options.baseDir;
+  this.scss = options.scss;
+}
+
+GenerateComponents.prototype.build = function() {
+  var components = componentsToGenerate(this.baseDir, this.scss);
+
+  components.forEach(function(hash) {
+    this.generateFile(hash.component, hash.default);
+  }, this);
+};
+
+GenerateComponents.prototype.generateFile = function(fileName, defaultFile) {
+  var exportFrom = this.registryType === 'self' ? defaultFile : fileName;
+  var module = exportFrom.replace(/(.+)\.js$/, this.moduleName + '/$1');
+  var component = "export { default } from '" + module + "';\n";
+
+  var outputFile = path.join(this.outputPath, this.prefix, fileName);
+  var outputDir = path.dirname(outputFile);
+
+  mkdirp.sync(outputDir);
+  fs.writeFileSync(outputFile, component, 'utf8');
+};
+
+var componentsToGenerate = function(baseDir, scss) {
+  var scssFiles = walkSync(baseDir, { globs: scss });
+
+  var expectedJsFiles = scssFiles.map(function(fileName) {
+    return fileName.replace(/^styles\/components\/(.+)\.scss/, 'components/$1.js');
```

Comment:
Not a bad idea, but maybe leave for when we cross that bridge? I'd want to be sure that it's configurable so that we can change the convention when we want to (`ui` might be a common enough prefix that we might want to think about something more unique.) Also, we may end up transforming the names of the files to make everything completely unique and provide helpers for component lookup (as another idea.)